challenge-manager/challenge-tree/inherited_timer.go

```diff
@@ -72,19 +73,19 @@ func (ht *RoyalChallengeTree) UpdateInheritedTimer(
 		if !ok {
 			return 0, fmt.Errorf("claimed edge %#x not found in royal tree", claimedEdgeId)
 		}
-		claimedEdgeInheritedTimer, err := ht.inheritedTimerForEdge(claimedEdge, blockNum)
+		claimedEdgeLocalTimer, err := ht.LocalTimer(claimedEdge, blockNum)
 		if err != nil {
 			return 0, err
 		}
-		if inheritedTimer > claimedEdgeInheritedTimer {
-			ht.inheritedTimers.Put(claimedEdge.Id(), claimedEdgeInheritedTimer)
-		}
-		onchainInheritedTimer, err := claimedEdge.InheritedTimer(ctx)
-		if err != nil {
-			return 0, err
+		claimedEdgeTimeUnrivaled := saturatingSum(protocol.InheritedTimer(claimedEdgeLocalTimer), inheritedTimer)
+		claimedEdgeInheritedTimer, ok := ht.inheritedTimers.TryGet(claimedEdge.Id())
+		if !ok {
+			claimedEdgeInheritedTimer = claimedEdgeTimeUnrivaled
+			ht.inheritedTimers.Put(edge.Id(), claimedEdgeInheritedTimer)
```

Comment:
we should be updating the claimed edge, not edge.Id()

Reply:
done.